node/src/service.rs

```diff
@@ -678,6 +689,227 @@ pub async fn start_parachain_node(
     .await
 }
 
+/// Start a solochain node.
+pub async fn start_solochain_node(
+    // Parachain config not used directly, but we need it to derive the default values for some container_config args
+    orchestrator_config: Configuration,
+    polkadot_config: Configuration,
+    mut container_chain_config: (ContainerChainCli, tokio::runtime::Handle),
+    collator_options: CollatorOptions,
+    hwbench: Option<sc_sysinfo::HwBench>,
+) -> sc_service::error::Result<(TaskManager, Arc<ParachainClient>)> {
+    let orchestrator_para_id = Default::default();
+    let parachain_config = prepare_node_config(orchestrator_config);
+    {
+        let (container_chain_cli, _) = &mut container_chain_config;
+        // If the container chain args have no --wasmtime-precompiled flag, use the same as the orchestrator
+        if container_chain_cli
+            .base
+            .base
+            .import_params
+            .wasmtime_precompiled
+            .is_none()
+        {
+            container_chain_cli
+                .base
+                .base
+                .import_params
+                .wasmtime_precompiled
+                .clone_from(&parachain_config.wasmtime_precompiled);
+        }
+    }
+
+    let chain_type: sc_chain_spec::ChainType = parachain_config.chain_spec.chain_type();
+    let relay_chain = crate::chain_spec::Extensions::try_get(&*parachain_config.chain_spec)
+        .map(|e| e.relay_chain.clone())
+        .ok_or("Could not find relay_chain extension in chain-spec.")?;
+
+    // Channel to send messages to start/stop container chains
+    let (cc_spawn_tx, cc_spawn_rx) = unbounded_channel();
+
+    // Create a `NodeBuilder` which helps setup parachain nodes common systems.
+    let mut node_builder = NodeConfig::new_builder(&parachain_config, hwbench.clone())?;
+
+    let (_block_import, import_queue) = import_queue(&parachain_config, &node_builder);
+
+    let (relay_chain_interface, collator_key) = node_builder
+        .build_relay_chain_interface(&parachain_config, polkadot_config, collator_options.clone())
+        .await?;
+
+    let validator = parachain_config.role.is_authority();
+
+    log::info!("start_solochain_node: is validator? {}", validator);
+
+    let node_builder = node_builder
+        .build_cumulus_network::<_, sc_network::NetworkWorker<_, _>>(
+            &parachain_config,
+            orchestrator_para_id,
+            import_queue,
+            relay_chain_interface.clone(),
+        )
+        .await?;
+
+    /*
+    let rpc_builder = {
```

Comment:
Why was this commented out?

Reply:
Because I wasn't sure if it's needed or not, will remove it now

Reply:
*fixed*